app_legacy/Helpers/database/player-rank.php

```diff
@@ -61,24 +61,28 @@ function recalculatePlayerPoints(string $user): bool
 
 function countRankedUsers(int $type = RankType::Hardcore): int
 {
-    $query = "SELECT COUNT(*) AS count FROM UserAccounts ";
-    switch ($type) {
-        case RankType::Hardcore:
-            $query .= "WHERE RAPoints >= " . Rank::MIN_POINTS;
-            break;
-
-        case RankType::Softcore:
-            $query .= "WHERE RASoftcorePoints >= " . Rank::MIN_POINTS;
-            break;
-
-        case RankType::TruePoints:
-            $query .= "WHERE TrueRAPoints >= " . Rank::MIN_TRUE_POINTS;
-            break;
-    }
-
-    $query .= " AND NOT Untracked";
-
-    return (int) legacyDbFetch($query)['count'];
+    return Cache::remember("rankedUserCount:$type",
+        60, // expire once a minute
```

Comment:
Surprisingly, this query accounted for almost half of the time in API_GetUserSummary on my dev machine (23/56ms). 

It's also used in the player rank section of the user page (twice - once for hardcore and once for softcore), so this should benefit all calls to any user page.